xfer/http_publisher.go

```diff
@@ -119,7 +119,9 @@ func (p HTTPPublisher) Publish(r io.Reader) error {
 }
 
 // Stop implements Publisher
-func (p HTTPPublisher) Stop() {}
+func (p *HTTPPublisher) Stop() {
+	p.client.Transport.(*http.Transport).CloseIdleConnections()
```

Comment:
Can you please add a short comment describing our debugging efforts this morning, so this important bit is not lost in a future refactor? :)